modules/data.land/R/clip_and_move_raster_file.R

```diff
@@ -0,0 +1,47 @@
+#' Clip and Move a Raster File
+#'
+#' Clips a raster to a polygon bounding box, optionally masks to polygon, and saves the 
+#' output in the same format as the input.
+#'
+#' @param input_path Character. Path to the input raster file.
+#' @param polygon An `sf` or `SpatVector` object to be used for clipping and masking.
+#' @param out_path Character. Path to save the processed raster.
+#' @return Character. The path to the saved output raster.
+#' @export
+#' @author David LeBauer
+clip_and_move_raster_files <- function(input_path, polygon, out_path, mask = TRUE, overwrite = TRUE) {
+  rast_in <- terra::rast(input_path)
+  # check that input file exists
+  if (!file.exists(input_path)) {
+    PEcAn.logger::logger.error("Input raster file does not exist: ", input_path)
+  }
+  # check that polygon is valid
+  if (!inherits(polygon, c("sf", "SpatVector"))) {
+    PEcAn.logger::logger.error("Polygon must be an sf object or SpatVector")
+  }
```

Comment:
I'm OK with this but can imagine cases (e.g. user passes path to a shapefile instead of an open handle) where we could go one step more flexible and accept anything coercible to SpatVector, letting any error on failure come from inside `terra::vect`.